client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import tinymce from 'tinymce/tinymce';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+const debug = debugFactory( 'calypso:tinymce-plugins:wpcom-paste-track' );
+
+const SOURCE_GOOGLE_DOCS = 'google_docs';
+const SOURCE_UNKNOWN = 'unknown';
+
+function trackPaste( editor ) {
+	debug( 'init' );
+
+	const store = editor.getParam( 'redux_store' );
+	if ( ! store ) {
+		debug( 'store not found, do not bind events' );
+		return;
+	}
+
+	/**
+	 * Records the event.
+	 *
+	 * @param {String} mode - 'html-editor' or 'visual-editor', indicates which editor was in use on paste.
+	 * @param {(Array|DOMStringList)} types - The types the content is available to paste.
```

Comment:
nice typing!

*(note that the `-` at the front of the description isn't necessary and it may end up actually a bit confusing for tools which interpret the JSDoc comments)